### PR TITLE
feat: cargo and poetry project packages support

### PR DESCRIPTION
**Pull request by hadrizi**
### Prerequisites

- [X] I have read and understood the `CONTRIBUTING` guide
- [X] The commit message follows the [conventional commits][cc] guidelines
- [X] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes/features)

### Description
see #2014 

cc @JanDeDobbeleer

### CODE REVIEW

File: src/segments/project.go [hunk 1]
```diff
@@ -24,6 +50,7 @@ type Project struct {
 
 	projects []*ProjectItem
 	Version  string
+	Name     string
```
Comment (JanDeDobbeleer) [inline src/segments/project.go]: I would create 1 type called ProjectData that contains these fields and is added using struct inheritance:

```go
type ProjectData struct {
    Name    string
    Version string
}

type Project struct {
    ...
 	projects []*ProjectItem
    ProjectData
```

You can then drop all project specific types, they don't really mean anything outside

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

File: src/segments/project.go (hunk 1 above)
Comment (hadrizi) [reply src/segments/project.go]: done

File: src/segments/project.go [hunk 2]
```diff
@@ -4,27 +4,43 @@ import (
...
+	Version string
+	Name    string
+}
+
+// Rust Cargo package
+type CargoTOML struct {
+	Package ProjectData
+}
+
+// Python Poetrey package
```
Comment (JanDeDobbeleer) [inline src/segments/project.go]: Typo -> `Poetry`

File: src/segments/project.go [hunk 3]
```diff
@@ -4,27 +4,43 @@ import (
...
+type PyprojectTOML struct {
```
Comment (JanDeDobbeleer) [inline src/segments/project.go]: I would respect the Pascal Casing: `PyProjectTOML`

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

**Comment by JanDeDobbeleer** [comment]
@hadrizi I will address the commitlint issue ;-)

File: src/segments/project.go (hunk 3 above)
Comment (hadrizi) [reply src/segments/project.go]: done

File: src/segments/project.go (hunk 2 above)
Comment (hadrizi) [reply src/segments/project.go]: done

**Review by JanDeDobbeleer — APPROVED** [review]

**Pull request merged**